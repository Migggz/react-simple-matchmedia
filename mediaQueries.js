export default {
  phone: '(min-width: 320px) and (max-width: 568px)',
  tablet: '(min-width : 768px) and (max-width : 1024px)',
  desktop: '(min-width : 1224px)',
};
