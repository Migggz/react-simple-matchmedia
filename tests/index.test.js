import React from 'react';
import { shallow } from 'enzyme';
import MediaQuery from '../index';
import mediaQueries from '../mediaQueries';

describe('MediaQuery', () => {
  it('Should render phone mediaQuery', () => {
    window.matchMedia = jest.fn().mockImplementation((query) => {
      return {
        addListener: jest.fn(),
        media: query,
        matches: true,
      };
    });

    const Component = shallow(<MediaQuery media="phone"><span>Hello MediaQuery</span></MediaQuery>);
    expect(window.matchMedia).toBeCalledWith(mediaQueries.phone);
    expect(Component).toMatchSnapshot();
  });

  it('Should not render desktop mediaQuery', () => {
    window.matchMedia = jest.fn(() => {
      return {
        addListener: jest.fn(),
        matches: false,
      };
    });

    const Component = shallow(<MediaQuery media="desktop">Hello MediaQuery</MediaQuery>);
    expect(window.matchMedia).toBeCalledWith(mediaQueries.desktop);
    expect(Component).toMatchSnapshot();
    expect(Component.state().matched).toBe(false);
  });

  it('Should work as render prop', () => {
    window.matchMedia = jest.fn(() => {
      return {
        addListener: jest.fn(),
        matches: true,
      };
    });

    const Component = shallow((
      <MediaQuery media="tablet">
        {matched => matched && <span>Lorem Ipsum</span>}
      </MediaQuery>
    ));

    expect(window.matchMedia).toBeCalledWith(mediaQueries.tablet);
    expect(Component.text()).toBe('Lorem Ipsum');
    expect(Component).toMatchSnapshot();
  });

  it('Should work with custom media query', () => {
    window.matchMedia = jest.fn(() => {
      return {
        addListener: jest.fn(),
        matches: true,
      };
    });

    const media = '(min-device-width : 768px) and (max-device-width : 1024px)';

    const Component = shallow((
      <MediaQuery media={media}>match me!</MediaQuery>
    ));

    expect(window.matchMedia).toBeCalledWith(media);
    expect(Component).toMatchSnapshot();
  });
});
