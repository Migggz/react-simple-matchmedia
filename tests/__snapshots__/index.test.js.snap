// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaQuery Should not render desktop mediaQuery 1`] = `""`;

exports[`MediaQuery Should render phone mediaQuery 1`] = `
<span>
  Hello MediaQuery
</span>
`;

exports[`MediaQuery Should work as render prop 1`] = `
<span>
  Lorem Ipsum
</span>
`;

exports[`MediaQuery Should work with custom media query 1`] = `"match me!"`;
